### ISSUE TITLE
Concatenating MS: Table column 'unknown' error

### ISSUE THREAD

**Issue by kendak333** [caracal-pipeline/caracal#1578]
I've processed two epochs of data on a field separately using caracal, up to selfcal. I'm trying to concatenate the files so that I can test additional calibration on the combined data before imaging. However when I try to do this using CASA's `concat` task, I get the following error:

`RuntimeError: Table column SIGMA_SPECTRUM is unknown`

Looking into CASA documentation, and it does seem to know that SIGMA_SPECTRUM is a valid column name. Got no help via them, so wondering if it's caracal-processing thing?

**Comment by KshitijT** [comment]
@kendak333 is this standalone casa outside caracal or is this the concat using the transform worker?

**Comment by KshitijT** [comment]
I am not sure if CASA should complain as long as both MSs have the same columns? In the initial reductions of the datasets (I am assuming with CARACal) did you switch on the specweights field in the prep worker?

**Comment by kendak333** [comment]
Yes - the extract from the `prep` worker below:

`specweights:`
`  enable:                       True`
`  mode:                         uniform`
`  calculate:`
`    statsfile:                    use_package_meerkat_spec`
`    weightcols:                   WEIGHT, WEIGHT_SPECTRUM`
`    noisecols:                    SIGMA, SIGMA_SPECTRUM`
`    apply:                        True`

I've just tried to concatenate two epochs observed during the same project (one day apart), and am getting the same error "SIGMA_SPECTRUM unknown" issue. Checking with a colleague who processed other epochs of the same set but have managed to concat without issue.

**Comment by paoloserra** [comment]
Hi @kendak333 

If I understand correctly, you're interested in concatenation along the time axis, which CARACal does not support. CARACal concatenates along the spectral axis only. See the description at https://caracal.readthedocs.io/en/latest/manual/workers/transform/index.html#concat .

I have a question though. Do you really need to concatenate? In CARACal, you should be able to image N input MS jointly (as long as the target is the same), and thus get a better sky model, which you can use to calibrate the individual MS files. Why do you need to concatenate them?

**Comment by paoloserra** [comment]
Yeah, I've had some pain with such minor metadata differences also wrt the channel width of FITS cubes -- not nice.

But back to the selfcal results, do you have any ideas about why joint imaging gives you worse artefacts? Was that in CARACal?

**Comment by kendak333** [comment]
It might be the case that they're there in (one of) the single-epoch images but below/near the noise, and the improvement in noise from the combination makes them pop up more starkly, though I'm not sure that'd explain the level I'm seeing - see below. The combined image is from sending the corrected data from both MSs into wsclean, no additional calibration applied.
![combined_image_test](https://github.com/caracal-pipeline/caracal/assets/33416498/424575c1-7079-4332-805a-bcd73d86ccc5)
Note no DD calibration has been applied at any stage so far.

**Comment by kendak333** [comment]
Definitely imaging CORRECTED_DATA (went and double checked!) - we're seeing this kind of behaviour quite a bit when we image multiple epochs together, which is why we're trying to go the selfcal-after-combining route.
